### PR TITLE
Caching changes

### PR DESCRIPTION
**Pull request by NeonMariia**

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -61,93 +63,280 @@ def mall_link(self):
...
-        LOG.info(str(request_lang))
-        LOG.info(user_request)
-        found, link = RequestHandler.existing_lang_check(request_lang, self.mall_link())
-        if found:
-            link = self.mall_link()+request_lang+'/directory/'
-            LOG.info('new link: '+ link)
-            return user_request, link
-        else:
-            self.speak_dialog("no_lang")
+        if message.data == {} or message is None:
```
Comment (NeonDaniel) [inline __init__.py:75]: Consider annotating the return params in a docstring

File: __init__.py [hunk 2]
```diff
@@ -61,93 +63,280 @@ def mall_link(self):
...
+            if open_time <= hour < close_time:
+                open_shops.append(shop)
+            elif day_time[1] == 'am' and open_time <= hour:
+                open_shops.append(shop)
+        return open_shops
+
+
+    def time_calculation(self, shop_info, open, day_time, hour, min):
+        # add logic if shop opens and closes not at am-pm time period
+        # change to speak dialog
```
Comment (NeonDaniel) [inline __init__.py:185]: Is this a TODO or just leftover comment? The docstring is good, so I think these lines can be removed

File: __init__.py [hunk 3]
```diff
@@ -61,93 +63,280 @@ def mall_link(self):
...
+        Returns:
+            3, None (to ask for another shop info)
+        Examples:
+            work time 9am-10pm
+            user's time 8am
+            Prompt: 'Shop is closed now. Opens in 1 hour'
+        """
+        for shop in shop_info:
+            work_time = shop['hours']
+            normalized_time = re.findall(r'(\d+)[am|pm]', work_time)
```
Comment (NeonDaniel) [inline __init__.py:219]: Just a note that using `datetime` might simplify things in the future; the parsing here looks good though

**Review by NeonDaniel — CHANGES_REQUESTED** [review]
.github directory should be replaced and __pycache__ directory removed. Otherwise changes look good

**Comment by NeonDaniel** [comment]
Is this related to https://github.com/NeonGeckoCom/skill-directory/pull/6 ?

**Comment by NeonMariia** [comment]
> Is this related to #6 ?

Yes, still related and can be merged I guess?

**Review by NeonDaniel — APPROVED** [review]

**Comment by NeonDaniel** [comment]
> > Is this related to #6 ?
> 
> Yes, still related and can be merged I guess?

Is this a replacement? This PR seems to include all of the commits from #6 and a few more

**Pull request merged**